packages/base-controller/CHANGELOG.md

```diff
@@ -7,27 +7,47 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [5.0.0]
+
 ### Added
 
-- Add and export type `StateConstraint`, which is an alias for `Record<string, Json>`, and represents the narrowest supertype encompassing the state of all controllers.
+- **BREAKING**: Add ESM build ([#3998](https://github.com/MetaMask/core/pull/3998))
+  - It's no longer possible to import files from `./dist` directly.
+- Add and export type `StateConstraint`, which is an alias for `Record<string, Json>` ([#3949](https://github.com/MetaMask/core/pull/3949))
+  - This type represents the narrowest supertype of the state of all controllers.
+  - Importing this type enables controllers to constrain state objects and types to be JSON-serializable without having to directly add `@metamask/utils` as a dependency.
 
 ### Changed
 
-- **BREAKING:** Narrow the generic constraint of the `ControllerState` parameter from `Record<string, unknown>` to `Record<string, Json>` for types `ControllerGetStateAction`, `ControllerStateChangeEvent`, `ControllerActions`, `ControllerEvents`.
-- **BREAKING:** Narrow the return type of the function `deriveStateFromMetadata<ControllerState>` from `Record<string, Json>` to `Record<keyof ControllerState, Json>`.
-- **BREAKING:** The `getRestricted` method of the `ControllerMessenger` class has major API changes that may require updates to its existing instances in downstream code. These changes align the type-level and runtime behavior of `getRestricted`, so that omitted or empty inputs consistently represent a set of empty allowlists. ([$4013](https://github.com/MetaMask/core/pull/4013))
-  - If the `AllowedActions`, `AllowedEvents` generic parameters are omitted, they are always assumed to be `never`.
-    - Previously, omission of these generic parameters resulted in the full allowlists for the controller being inferred as type constraints for the `allowedActions`, `allowedEvents` function parameters.
-  - If the function parameters `allowedActions`, `allowedEvents` are a non-empty array, their corresponding type names must be explicitly passed into generic parameters `AllowedActions`, `AllowedEvents` to avoid type errors.
+- **BREAKING:** Narrow the return type of the function `deriveStateFromMetadata<ControllerState>` from `Record<string, Json>` to `Record<keyof ControllerState, Json>` ([#3949](https://github.com/MetaMask/core/pull/3949))
+  - TODO: What's the result of this change? `deriveStateFromMetadata` is not public, but `getAnonymizedState` and `getPersistentState` are.
```

Comment:
With #4040 merged, this could be our new entry:

```suggestion 
- **BREAKING:** Narrow the return types of functions `getAnonymizedState<ControllerState>` and `getPersistentState<ControllerState>` from `Record<string, Json>` to `Record<keyof ControllerState, Json>`. ([#3949](https://github.com/MetaMask/core/pull/3949), [#4040](https://github.com/MetaMask/core/pull/4040))
```